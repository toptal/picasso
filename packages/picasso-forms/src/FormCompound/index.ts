import { FieldRequirements } from '@toptal/picasso'

import Form from '../Form'
import Autocomplete from '../Autocomplete'
import Input from '../Input'
import Select from '../Select'
import Radio from '../Radio'
import ButtonRadio from '../ButtonRadio'
import RadioGroup from '../RadioGroup'
import Checkbox from '../Checkbox'
import ButtonCheckbox from '../ButtonCheckbox'
import CheckboxGroup from '../CheckboxGroup'
import NumberInput from '../NumberInput'
import FileInput from '../FileInput'
import DatePicker from '../DatePicker'
import TimePicker from '../TimePicker'
import TagSelector from '../TagSelector'
import SubmitButton from '../SubmitButton'
import Switch from '../Switch'
import Rating from '../Rating'
import Dropzone from '../Dropzone'
import RichTextEditor from '../RichTextEditor'
import PasswordInput from '../PasswordInput'
import { FormConfigContext } from '../FormConfig'
import AvatarUpload from '../AvatarUpload'

type FormCompoundType = typeof Form & {
  Autocomplete: typeof Autocomplete
  Input: typeof Input
  Select: typeof Select
  Radio: typeof Radio
  ButtonRadio: typeof ButtonRadio
  RadioGroup: typeof RadioGroup
  Checkbox: typeof Checkbox
  ButtonCheckbox: typeof ButtonCheckbox
  CheckboxGroup: typeof CheckboxGroup
  NumberInput: typeof NumberInput
  FileInput: typeof FileInput
  DatePicker: typeof DatePicker
  TimePicker: typeof TimePicker
  TagSelector: typeof TagSelector
  SubmitButton: typeof SubmitButton
  ConfigProvider: typeof FormConfigContext.Provider
  Switch: typeof Switch
  Rating: typeof Rating
  Dropzone: typeof Dropzone
  PasswordInput: typeof PasswordInput
  FieldRequirements: typeof FieldRequirements
  RichTextEditor: typeof RichTextEditor
  AvatarUpload: typeof AvatarUpload
}

export const FormCompound: FormCompoundType = Object.assign(Form, {
  Autocomplete: Autocomplete,
  Input: Input,
  Select: Select,
  Radio: Radio,
  ButtonRadio: ButtonRadio,
  RadioGroup: RadioGroup,
  Checkbox: Checkbox,
  ButtonCheckbox: ButtonCheckbox,
  CheckboxGroup: CheckboxGroup,
  NumberInput: NumberInput,
  FileInput: FileInput,
  DatePicker: DatePicker,
  TimePicker: TimePicker,
  TagSelector: TagSelector,
  SubmitButton: SubmitButton,
  ConfigProvider: FormConfigContext.Provider,
  Switch: Switch,
  Rating: Rating,
  Dropzone: Dropzone,
  PasswordInput: PasswordInput,
  FieldRequirements: FieldRequirements,
  RichTextEditor: RichTextEditor,
  AvatarUpload: AvatarUpload,
})
