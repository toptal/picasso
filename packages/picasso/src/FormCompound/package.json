{
  "name": "@toptal/picasso-form-compound",
  "version": "1.0.0",
  "main": "index.js",
  "dependencies": {
    "@toptal/picasso-form": "^1.0.0",
    "@toptal/picasso-form-error": "^1.0.0",
    "@toptal/picasso-form-field": "^1.0.0",
    "@toptal/picasso-form-hint": "^1.0.0",
    "@toptal/picasso-form-label": "^1.0.0"
  }
}