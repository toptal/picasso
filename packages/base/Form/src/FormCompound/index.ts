import { Form } from '../Form'
import { FormError } from '../FormError'
import { FormField } from '../FormField'
import { FormHint } from '../FormHint'
import { FormLabel } from '../FormLabel'

type FormCompoundType = typeof Form & {
  Field: typeof FormField
  Hint: typeof FormHint
  Error: typeof FormError
  Label: typeof FormLabel
}

export const FormCompound: FormCompoundType = Object.assign(Form, {
  Field: FormField,
  Hint: FormHint,
  Error: FormError,
  Label: FormLabel,
})
