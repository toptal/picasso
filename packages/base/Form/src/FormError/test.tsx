import React from 'react'
import type { RenderResult } from '@toptal/picasso-test-utils'
import { render } from '@toptal/picasso-test-utils'
import type { OmitInternalProps } from '@toptal/picasso-shared'
import { Input } from '@toptal/picasso-input'

import type { Props } from './FormError'
import { FormCompound as Form } from '../FormCompound'

const TestFormError = ({ children }: OmitInternalProps<Props>) => {
  return (
    <Form>
      <Form.Field>
        <Form.Label>Label:</Form.Label>
        <Input />
        <Form.Error>{children}</Form.Error>
        <Form.Hint>This is a hint</Form.Hint>
      </Form.Field>
    </Form>
  )
}

describe('FormError', () => {
  let api: RenderResult

  beforeEach(() => {
    api = render(<TestFormError>My Error</TestFormError>)
  })
  it('renders', () => {
    const { container } = api

    expect(container).toMatchSnapshot()
  })
})
