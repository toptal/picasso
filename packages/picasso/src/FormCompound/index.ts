import Form from '../Form'
import FormError from '../FormError'
import FormField from '../FormField'
import FormHint from '../FormHint'
import FormLabel from '../FormLabel'

export const FormCompound = Object.assign(Form, {
  Field: FormField,
  Hint: FormHint,
  Error: FormError,
  Label: FormLabel,
})
