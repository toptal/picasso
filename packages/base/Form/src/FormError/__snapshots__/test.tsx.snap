// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormError renders 1`] = `
<div>
  <div
    class="Picasso-root"
  >
    <form
      class=""
    >
      <div
        class="PicassoFormField-root"
        data-field-has-error="false"
      >
        <label
          class="PicassoFormLabel-root"
        >
          <span
            class="PicassoFormLabel-medium"
          >
            Label:
          </span>
        </label>
        <div
          class="base-Input base-Input base-Input relative inline-flex gap-y gap-x items-center rounded-sm [font-size:_unset] hover:[&_.resetButtonDirty]:visible p-2 h-8 w-[18.75rem] bg-white after:content-[""] after:inline-block after:absolute after:top-0 after:bottom-0 after:right-0 after:left-0 after:pointer-events after:border-solid after:rounded-sm after:border-gray [&:has(input:focus)]:after:border-blue after:border [&:has(input:focus)]:after:shadow-[0_0_0_3px] [&:has(input:focus)]:after:shadow-blue hover:[&:not(:has(input:focus))]:after:border-gray text-black cursor-[inherit]"
        >
          <input
            aria-invalid="false"
            autocomplete="none"
            class="base-Input [&::-webkit-calendar-picker-indicator]:bg bg-transparent border-none box-border cursor-[inherit] h-full outline-none p-0 peer resize-none w-full text-md leading-4 text-black [&::placeholder]:text-gray [&::placeholder]:opacity-100"
            type="text"
            value=""
          />
        </div>
        <div
          class="PicassoFormError-root"
        >
          <p
            class="m-0 text-2xs text-red font-regular cursor-default"
          >
            My Error
          </p>
        </div>
        <div
          class="FormHint-root"
        >
          <p
            class="m-0 text-2xs text-graphite font-regular"
          >
            This is a hint
          </p>
        </div>
      </div>
    </form>
  </div>
</div>
`;
